ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1030,13 +1037,13 @@ public void write()
           hardDeleter.preLogFlush();
 
           // flush the log to ensure everything till the fileEndPointerBeforeFlush is flushed
-          log.flush();
+          logSegment.flush();
```

Comment:
Would need to flush more than this segment in the general case, in case rollover happened twice in the flush interval. Wouldn't you?